kafka.cpp

```diff
@@ -36,6 +38,31 @@ static void dr_msg_cb(rd_kafka_t *rk, const rd_kafka_message_t *rkmessage, void
 	}
 }
 
+/**
+ *  Callback to handle errors
+ */
+static void error_cb(rd_kafka_t *rk, int err, const char *reason, void *opaque)
+{
+    rd_kafka_resp_err_t kafkaError = (rd_kafka_resp_err_t) err;
+	switch (kafkaError)
+	{
+		case RD_KAFKA_RESP_ERR__TRANSPORT:
+			m_error = true;
+			Logger::getLogger()->error("Kafka : %s : %s ", rd_kafka_err2str(kafkaError), reason );
+			break;
+		case RD_KAFKA_RESP_ERR__ALL_BROKERS_DOWN:
+			m_error = true;
+			Logger::getLogger()->error("Kafka : %s : %s ", rd_kafka_err2str(kafkaError), reason );
+			break;
+		case RD_KAFKA_RESP_ERR__RESOLVE:
+			Logger::getLogger()->warn("Kafka : %s : %s ", rd_kafka_err2str(kafkaError), reason );
+			break;
+		default:
+			Logger::getLogger()->error("Kafka : %s : %s ", rd_kafka_err2str(kafkaError), reason );
+			break;
+	}
+}
```

Comment:
After processing error, m_error should be reset to false again, here or before the next transmission, is that done somewhere?

Reply:
Fixed